### DISCUSSION TITLE
What is considered best practice for storing JS function in MongoDB?

### DISCUSSION THREAD

**Discussion by tal-rofe** [prisma/prisma#16059] [Q&A] [answered]
I want to store a user input function into the database.
Obviously I could store that input as a string in the database.

But I wonder whether there is a better way to do so?

I just saw something called `Code` class in MongoDB: https://mongodb.github.io/node-mongodb-native/api-bson-generated/code.html

Is there any better way to do it using Prisma?

**Comment by nurul3101** [comment, marked as answer]
Hey @tal-rofe 👋 

Did you have a look at this section of [Storing JavaScript Function on Server in MongoDB](https://www.mongodb.com/docs/manual/tutorial/store-javascript-function-on-server/), Would this fit your use case? Do you plan to invoke these stored functions against your database?
Please do note that storing application logic in the database is not encouraged.

For invoking those stored functions you would need to use [runCommandRaw](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#runcommandraw).

**Reply by tal-rofe** [reply]
> Hey @tal-rofe 👋 
> 
> 
> 
> Did you have a look at this section of [Storing JavaScript Function on Server in MongoDB](https://www.mongodb.com/docs/manual/tutorial/store-javascript-function-on-server/), Would this fit your use case? Do you plan to invoke these stored functions against your database?
> 
> Please do note that storing application logic in the database is not encouraged.
> 
> 
> 
> For invoking those stored functions you would need to use [runCommandRaw](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#runcommandraw).
> 
> 

Actually I just need to store snippet of code. Not necessarily a function

**Reply by nurul3101** [reply]
Do you plan to invoke the function? If not, I think storing it as a String would be the most straightforward way.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
